src/core/server/integration_tests/ui_settings/doc_missing.ts

```diff
@@ -13,6 +13,8 @@ export const docMissingSuite = (savedObjectsIndex: string) => () => {
   beforeEach(async () => {
     const { esClient } = getServices();
 
+    await esClient.indices.refresh({ index: savedObjectsIndex });
```

Comment:
`esClient.deleteByQuery` below started failing with conflict. see for more details https://github.com/elastic/kibana/issues/159662#issuecomment-1621441238

Reply:
Looking at the `src/core/server/integration_tests/ui_settings/index.test.ts`, at first glance it would seem that this test logic can be greatly simplified:
* We could simplify the `index.test.ts` and remove the `savedObjectsIndex` and `kibanaVersion` constants.
* We could obtain the esArchiver service, and call `esArchiver.emptyKibanaIndex()`, which you are also updating in this PR. This way we would already be doing a cleanup + refresh. WDYT?

But then, I wonder why you are forced to do the refresh before the delete. 🤔 
The first error you mention seems to be related to `version_conflicts":1`.

Reply:
> We could simplify the index.test.ts and remove the savedObjectsIndex and kibanaVersion constants.
> We could then obtain the esArchiver service in this file, and call esArchiver.emptyKibanaIndex(), which you are also updating in this PR. This way we would already be doing a cleanup + refresh. WDYT?


We took a look at this with @gsoldevila. It turns this isn't as straightforward as it seemed as esArchiver doesn't exist in integration tests and we decided it isn't worth to try to init it there